File: year2021/day01/python/solution.py
def main():
    print(part_1())
    print(part_2())


def part_1():
    return sum([b > a for a, b in zip(input, input[1:])])


def part_2():
    return sum(sum(input[i+1:i+4]) > sum(input[i:i+3]) for i in range(len(input)-2))


def setup(path):
    global input
    
    with open(path, 'r') as file:
        input = [int(row) for row in file]


if __name__ == '__main__':
    setup('../puzzle.txt')
    main()

File: year2021/day01/java/Solution.java
import java.nio.file.*;
import java.util.*;

public class Solution {
    private static int[] NUMBERS;

    public static void main(String[] args) { 
        try {
            setup(Files.readString(Paths.get("../puzzle.txt")));
            System.out.println(part1());
            System.out.println(part2());
        } catch(Exception e) {
            e.printStackTrace();
        }
    }
    
	public static Integer part1() {
        int sum = 0;
        for (int i = 0; i < NUMBERS.length - 1; i++) {
            if (NUMBERS[i + 1] > NUMBERS[i]) {
                sum++;
            }
        }
        return sum;
    }
    
    public static Integer part2() {
        int count = 0;
        for (int i = 0; i < NUMBERS.length - 2; i++) {
            if (sum(NUMBERS, i + 1, i + 4) > sum(NUMBERS, i, i + 3)) {
                count++;
            }
        }
        return count;
    }

    private static int sum(int[] array, int start, int end) {
        int sum = 0;
        for (int i = start; i < end && i < array.length; i++) {
            sum += array[i];
        }
        return sum;
    }

    public static void setup(String input) {
        NUMBERS = Arrays.stream(input.split("\n")).mapToInt(Integer::parseInt).toArray();
    }
}